Sharpasonne/Engine.cs

```diff
@@ -35,23 +46,27 @@ public class Engine : IEngine
             {
                 var typeNames = string.Join(", ", nonGameActions.Select(t => t.FullName));
                 var message = $"'{typeNames}' does not implement {nameof(IGameAction)}";
-                throw new ArgumentOutOfRangeException(message);
+                return Option.None<Engine, Exception>(new ArgumentOutOfRangeException(nameof(gameActions), message));
```

Comment:
Return an exception as an error? Why not a string?

Reply:
See [above](https://github.com/overengineering/sharpasonne/pull/15/files/4b1da4e71227408c064a2512201e365ff7c8c4d6#r170530605)